meshroom/core/__init__.py

```diff
@@ -94,6 +104,76 @@ def loadPlugins(folder, packageName, classType):
     return pluginTypes
 
 
+def validateNodeDesc(nodeDesc):
+    """
+    Check that the node has a valid description before being loaded. For the description
+    to be valid, the default value of every parameter needs to correspond to the type
+    of the parameter. An empty returned list means that every parameter is valid, and so
+    is the node's description.
+
+    Args:
+        nodeDesc (desc.Node): description of the node
+
+    Returns:
+        errorList (list): the list of invalid parameters if there are any, empty list otherwise
+    """
+    errorList = []
+
+    def validateBasicTypes(param, errors):
+        # Validate base types (int, float, string, bool)
+        # ChoiceParam and File are not checked
+        if isinstance(param, desc.IntParam) and (not isinstance(param.value, int)
```

Comment:
`param.validateValueType(param.value)`
It should be part of each XXParam class. We should not maintain lists of all known types like this.

Reply:
Done